06-Time/For_kernel_space/module_cpustat/module_cpustat.c

```diff
@@ -0,0 +1,148 @@
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/device.h>
+#include <linux/err.h>
+
+#include <linux/fs.h>
+#include <asm/segment.h>
+#include <asm/uaccess.h>
+#include <linux/buffer_head.h>
+
+#include <linux/timer.h>
+
+unsigned char arr[4096];
+struct timer_list mytimer;
+int currCpuLoad = 0;
+
+// file work routines
+struct file *file_open(const char *path, int flags, int rights) 
+{
+    struct file *filp = NULL;
+    mm_segment_t oldfs;
+    int err = 0;
+
+    oldfs = get_fs();
+    set_fs(get_ds());
+    filp = filp_open(path, flags, rights);
+    set_fs(oldfs);
+    if (IS_ERR(filp)) {
+        err = PTR_ERR(filp);
+        return NULL;
+    }
+    return filp;
+}
+
+int file_read(struct file *file, unsigned long long offset, unsigned char *data, unsigned int size) 
+{
+    mm_segment_t oldfs;
+    int ret;
+
+    oldfs = get_fs();
+    set_fs(get_ds());
+
+    ret = vfs_read(file, data, size, &offset);
+
+    set_fs(oldfs);
+    return ret;
+}   
+
+void file_close(struct file *file) 
+{
+    filp_close(file, NULL);
+}
+
+// get cpu average load routine
+int getCpuStat(void)
+{
+	struct file *myfile = file_open("/proc/stat", O_RDONLY, 0);
+	file_read(myfile, 0, arr, 4096); 
```

Comment:
Implemented function file_open() can return NULL value. Please consider this flow program flow.